add intent to retain to claims query in DCQL (#338)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

If the `values` property is present, the Wallet SHOULD return the claim only if the
type and value of the claim both match for at least one of the elements in the array. Details of the processing rules are defined in (#selecting_claims).

`intent_to_retain`

> : OPTIONAL. A boolean that indicates the Verifier intends to retain the Claim's data that is being requested.

Reviewer: ```suggestion
: OPTIONAL. A boolean that indicates the Verifier intends to store the Claim's data that is being requested beyond the processing of the response, for example, for purposes of recording a proof of the transaction or to maintain a user profile.
```
Not sure if this is the best wording, but we should try to explain what the 'retain' in `intent_to_retain` means.

If the `values` property is present, the Wallet SHOULD return the claim only if the
type and value of the claim both match for at least one of the elements in the array. Details of the processing rules are defined in (#selecting_claims).

`intent_to_retain`

> : OPTIONAL. A boolean that indicates the Verifier intends to store the Claim's data that is being requested beyond the processing of the response. For example, to meet regulatory requirements of retaining records of transactions or to maintain user profiles. If omitted, the default value is `false`.

Author: ```suggestion
: OPTIONAL. A boolean that indicates the Verifier intends to store the Claim's data that is being requested beyond the processing of the response. For example, to meet regulatory requirements of retaining records of transactions or to maintain user profiles.
```

Author: + move it to the mdoc section as a starting point.

doctype of the requested Verifiable Credential. It MUST
be a valid doctype identifier as defined in [@ISO.18013-5].

#### Parameters in the Claims Query {#mdocs_claims_query}

The following are ISO mdoc specific parameters to be used in a Claims Query as defined in (#claims_query).

> `namespace`:
> : REQUIRED if the Credential Format is based on the mdoc format defined in [@ISO.18013-5]; MUST NOT be present otherwise. 
> The value MUST be a string that specifies the namespace of the data element
> within the mdoc, e.g., `org.iso.18013.5.1`.
> 
> `claim_name`:
> : REQUIRED if the Credential Format is based on mdoc format defined in [@ISO.18013-5]; MUST NOT be present otherwise. 
> The value MUST be a string that specifies the data element identifier of the data element within the provided namespace
> in the mdoc, e.g., `first_name`.
> 

Reviewer 2: These parameters were removed as part of https://github.com/openid/OpenID4VP/pull/368 that has been merged

```suggestion
```
